### PR TITLE
thandle needs an accessor to t

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: inc/azure_c_util/thandle.h [hunk 1]
```diff
@@ -12,7 +12,10 @@
 #include "refcount_os.h"
 
 /*the incomplete unassignable type*/
-#define THANDLE(T) const T* const
+#define THANDLE(T) MU_C2(CONST_P2_CONST_,T)
```
Comment (dcristoloveanu) [inline inc/azure_c_util/thandle.h]: >T) MU_C2(CONST_P2_CONST_,T) [](start = 16, length = 27)

Ah, this should fix the issue I mentioned with passing the array of handles as out arg.

Comment (dcristoloveanu) [reply inc/azure_c_util/thandle.h]: Nice.

---
In reply to: [392014336](https://github.com/Azure/azure-c-util/pull/20#discussion_r392014336) [](ancestors = 392014336)

File: tests/thandle_ut/thandle_ut.c [hunk 2]
```diff
@@ -488,4 +488,41 @@ TEST_FUNCTION(THANDLE_FREE_with_t_NULL_returns)
     ///cleanup
 }
```
Comment (dcristoloveanu) [inline tests/thandle_ut/thandle_ut.c]: Can you actually add a test for:

THANDLE(T)** x;
*x = malloc(sizeof(THANDLE(T)) * 42);

I suspect it should compile now.

Comment (anporumb) [reply tests/thandle_ut/thandle_ut.c]: Wrote a test

---
In reply to: [392014690](https://github.com/Azure/azure-c-util/pull/20#discussion_r392014690) [](ancestors = 392014690)

**Pull request merged**